src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordObjEquals.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal sealed class SynthesizedRecordObjEquals : SynthesizedInstanceMethodSymbol
```

Comment:
What about synthesizing an implementation of `IEqualityComparer<ThisType>` or `IEqualityComparer<BaseType>`?

Reply:
Right now this doesn't synthesize any IEquatable implementations or the `==` operators. I'll leave that open for later.